### ISSUE TITLE
Improve Arcade build

### ISSUE THREAD

**Issue by mstv** [gitextensions/gitextensions#7702]
### Tests

- [x] Do not run tests in parallel -- as it seems to be:
  ```
  Running tests: C:\projects\gitextensions\artifacts\bin\tests\UnitTests\GitCommands.Tests\net461\GitCommands.Tests.dll [net461|x64]
  Running tests: C:\projects\gitextensions\artifacts\bin\tests\UnitTests\GitUI.Tests\net461\GitUI.Tests.dll [net461|x64]
  Tests succeeded: C:\projects\gitextensions\artifacts\bin\tests\UnitTests\GitCommands.Tests\net461\GitCommands.Tests.dll [net461|x64]
  ```
- [x] On build fail, the AppVeyor build shall fail (turn red)
- [x] On test fail, the AppVeyor build shall fail (turn red)
  - [x] after running all tests
  - [x] after storing the artifacts
  - [x] including each `TestResult.xml`
- [x] Show failed test cases on the AppVeyor console (e.g. just `type FooTest.log`)
- [ ] All test passed in about 3 minutes before. (I don't know the reason for this speedup a few weeks ago.)
  With the Arcade build, the test run takes about 10 minutes although tests are run in parallel.
- [ ] Double check all files matching `tests/.*\.txt` for correct line endings.
- [ ] There should be a default active project, which is started on Run/Debug (`F5`).
      STR: `git clean -xdf`; open sln; hit `F5` => does not build and run
- [x] Check `English.xlf` (only in CI build).
- [x] `Build/Run-Tests.ps1` has become obsolete - remove
- [x] Are there more obsolete scripts?

**Comment by RussKie** [comment]
> * `Build/Run-Tests.ps1` has become obsolete - remove
> * Are there more obsolete scripts?

Yes, quite likely. There is more work to be done to complete the conversion.

**Comment by RussKie** [comment]
> * Do not run tests in parallel

Tests _are_ run in sequence.
https://github.com/gitextensions/gitextensions/blob/0b00208914c1e126de69fdbf162bc2252113b344/Build/tools/NUnit/NUnit.targets#L44
You can verify it by watching at Process Explorer while executing tests - there's only one running at a time. You can also check the binlog (look under .\gitextensions\artifacts\log\[Debug|Release], use [MSBuildStructuredLog](https://github.com/KirillOsenkov/MSBuildStructuredLog)) and see what command line is invoked:
![image](https://user-images.githubusercontent.com/4403806/73445601-28bc7a80-43af-11ea-8020-992b4238279a.png)

The order of the output is for the most part irrelevant - the build is run concurrently, however the test runner is synchronises itself.

In fact I propose in https://github.com/gitextensions/gitextensions/pull/7703/commits/274a85e47809797e54ca4349aa46b3862f271b44 that unit tests are run in parallel as by definition they should be independent and self contained. Any test that doesn't fit into this description should be converted to an integration test and moved to an integration tests project.

**Comment by RussKie** [comment]
> * Show build progress in the `Output` pane also when triggered by Run/Debug (`F5`)

Not sure what you mean. The output is shown when the solution is built:
![image](https://user-images.githubusercontent.com/4403806/73446034-24dd2800-43b0-11ea-8d3a-218f8586dc02.png)
The verbosity has been set to `minimal` by default.

If you wish to have a more detailed output, dial it up in the settings:
![image](https://user-images.githubusercontent.com/4403806/73446225-7ab1d000-43b0-11ea-9946-1427cf8dfeeb.png)

**Comment by RussKie** [comment]
> * On build fail, the AppVeyor build shall fail (turn red)

Not quite sure why the build isn't getting failed.
I'll have to look into it.

> * Show failed test cases on the AppVeyor console (e.g. just `type FooTest.log`)

Test results are still available in the [Tests] tab: https://ci.appveyor.com/project/gitextensions/gitextensions/builds/30456429/tests
Test logs are published as artifacts: https://ci.appveyor.com/project/gitextensions/gitextensions/builds/30456429/artifacts

I don't see a compelling reason to show logs for passed tests, but I suppose we can dump a test log to the console if tests fail. 
I'll have to look into it.

**Comment by mstv** [comment]
> Tests _are_ run in sequence.

It looks like the tests in
- UnitTests\Plugins\DeleteUnusedBranches.Tests\,
- IntegrationTests\UI.IntegrationTests\ and
- UnitTests\GitUI.Tests\

do run in parallel here:

![grafik](https://user-images.githubusercontent.com/36601201/73490550-0d507000-43ad-11ea-8131-1b3b9b7a68c7.png)

**Comment by mstv** [comment]
> Any test that doesn't fit into this description should be converted to an integration test and moved to an integration tests project.

I agree.

> In fact I propose in 274a85e that unit tests are run in parallel as by definition they should be independent and self contained. 

This comes much too early. We have several singletons, at least `AppSettings` and clipboard.
First the tests must pass stably. Then we can try to run them in parallel and handle the issues.

**Comment by mstv** [comment]
> > * Show failed test cases on the AppVeyor console (e.g. just `type FooTest.log`)

> Test results are still available in the [Tests] tab: https://ci.appveyor.com/project/gitextensions/gitextensions/builds/30456429/tests

I don't trust the [Tests] tab anymore. Some tests are labeled with different color (yellow blinking, gray) although they have passed and have not printed to the console (lightblue text), e.g. 
`GitCommandsTests.GitModuleTests.BatchUnstageFiles_should_work_as_expected([abc2, abc2, a` and `GitCommandsTests.GitModuleTests.GetSuperprojectCurrentCheckout` in [build](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/30420842/tests). BTW: no response to the regarding bugreport for moths.
And it is not made for running the same tests multiple times.

> Test logs are published as artifacts: https://ci.appveyor.com/project/gitextensions/gitextensions/builds/30456429/artifacts

I know. But we should store the XML result files, too.
My main point is: It looks like MSBuild stops when a test assembly reported errors.

**Comment by RussKie** [comment]
> It looks like the tests in
> 
> * UnitTests\Plugins\DeleteUnusedBranches.Tests,
> * IntegrationTests\UI.IntegrationTests\ and
> * UnitTests\GitUI.Tests\
> 
> do run in parallel here:

Unit tests and integration tests can run in parallel.
If you don't want to run them in parallel then we need to break their run to:
```
.\build.cmd -test
.\build.cmd -integrationTest
```
The same goes out to the CI build:
```
& .\cibuild.cmd -test
& .\cibuild.cmd -integrationTest
```

**Comment by RussKie** [comment]
> I don't trust the [Tests] tab anymore

Time to move to AzDo :)

**Comment by pmiossec** [comment]
@RussKie Are we aware that the builds no more generate zip version as build artifacts?
And so, that the download link bring nowhere?

![image](https://user-images.githubusercontent.com/460196/73936695-ae768380-48e3-11ea-9bca-1df64f156689.png)

**Comment by RussKie** [comment]
Yes, the artifacts publishing is currently disabled.
It is on my todo list. I've flat out busy with work lately.

**Comment by RussKie** [comment]
@mstv is there anything outstanding from the list?

**Comment by mstv** [comment]
> is there anything outstanding from the list?

😉 everything unticked in the issue description (The "random" `F5` issue can be ignored.)
and of course the installer as well as the `portable.zip`.

**Comment by gerhardol** [comment]
When I debug locally, my changes are not in effect sometimes, I have to build a second time.
(not plugins).
Is this related?

**Comment by RussKie** [comment]
> When I debug locally, my changes are not in effect sometimes, I have to build a second time.

It is hard to tell, there are lot of moving parts in the build process.
If you provide more details it may be easier to hypothesise.

**Comment by RussKie** [comment]
> and of course.... as the `portable.zip`.

There: https://ci.appveyor.com/project/gitextensions/gitextensions/builds/30847034/artifacts
The app only, the plugins are yet to be bundled

**Comment by mstv** [comment]
Am I the only one with this popup on the first run of a portable build?
![grafik](https://user-images.githubusercontent.com/36601201/74687437-afa58b80-51d4-11ea-9863-23bf34f2289f.png)
I think we must reopen #7571, @NikolayXHD.

**Comment by NikolayXHD** [comment]
> Am I the only one with this popup on the first run of a portable build?
> I think we must reopen #7571, @NikolayXHD.

no, not the only one:
https://github.com/gitextensions/gitextensions/pull/7785#issuecomment-586936813

#7562 should fix it

**Comment by mstv** [comment]
What about `English.xlf`?
I guess the regarding check should be re-added to `AppVeyor.yml`.

**Comment by RussKie** [comment]
Yes, one step a time :)

**Comment by RussKie** [comment]
I don't believe there is anything outstanding.

**Comment by mstv** [comment]
> I don't believe there is anything outstanding.

> Double check all files matching tests/.*\.txt for correct line endings

The line endings of some test files, e.g. `UnitTests/GitUI.Tests/Editor/FileViewerTextTests.DoAutoCRLF*.txt`, have changed with commit 0b00208914c1e126de69fdbf162bc2252113b344.
I am slightly surprised and concerned that it has no influence on the tests. They do pass.
Just confirm that this is OK or even intended. Then this issue can be closed.

**Comment by mstv** [comment]
The line endings of some test files, e.g. `UnitTests/GitUI.Tests/Editor/FileViewerTextTests.DoAutoCRLF*.txt`, have changed with commit 0b00208.
I am slightly surprised and concerned that it has no influence on the tests. They do pass on AppVeyor (also refer to #7417).
@RussKie, just confirm that this change is OK or even intended. Then this issue could be closed.

**Comment by RussKie** [comment]
I tried reverting the files, but the only difference I got that these tests started failing intermittently...
¯\\_(ツ)\_/¯

**Comment by RussKie** [comment]
Raised https://github.com/gitextensions/gitextensions/issues/8000 to track the issue.

**Comment by RussKie** [comment]
I was aware of the fact that initially ApprovalTests considered line endings, and it caused troubles. So at some point they [stopped considering EoL differences as fatal](https://github.com/approvals/ApprovalTests.Net#approved-file-artifacts). 
I think as part of the migration to the Arcade SDK I must have updated to the new-ish version of ApprovalTests.

But until now I failed to make a connection... 🤕 This explains why the tests no longer fail, though the line endings have been changed.
